step-functions-plugins/step-functions-plugins-jmeter/step-functions-plugins-jmeter-def/src/test/java/step/plugins/jmeter/JMeterHandlerTest.java

```diff
@@ -1,84 +1,85 @@
- *  
+ *
  * You should have received a copy of the GNU Affero General Public License
  * along with STEP.  If not, see <http://www.gnu.org/licenses/>.
  ******************************************************************************/
 package step.plugins.jmeter;
 
-import java.io.File;
-import java.io.IOException;
-import java.util.HashMap;
-import java.util.Map;
-
+import ch.exense.commons.app.Configuration;
 import jakarta.json.JsonObject;
-
 import org.bson.types.ObjectId;
 import org.junit.Assert;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
-import ch.exense.commons.app.Configuration;
 import step.core.accessors.AbstractOrganizableObject;
 import step.core.dynamicbeans.DynamicValue;
 import step.functions.io.Output;
 import step.functions.runner.FunctionRunner;
 import step.functions.runner.FunctionRunner.Context;
 import step.grid.bootstrap.ResourceExtractor;
 
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
 public class JMeterHandlerTest {
-	
-	private static final Logger logger = LoggerFactory.getLogger(JMeterHandlerTest.class);
-	
-	@Test 
-	public void test1() {
-		File jmeterHomer = new File("../../distribution/template-controller/ext/jmeter");
-		if(jmeterHomer.exists()) {
-			JMeterFunction f = buildTestFunction();
-			Output<JsonObject> output = run(f, "{}");
-			Assert.assertNull(output.getError());
-			Assert.assertNotNull(output.getPayload().get("samples"));
-		} else {
-			logger.warn("Skipping JMeterHandler test as no JMeter installation could been found");
-		}
-	}
-	private Output<JsonObject> run(JMeterFunction f, String inputJson) {
-		Configuration configuration = new Configuration();
-		configuration.putProperty("plugins.jmeter.home", "../../distribution/template-controller/ext/jmeter");
-		
-		try (Context context = FunctionRunner.getContext(configuration,new JMeterFunctionType(configuration), new HashMap<>())) {
-			return context.run(f, inputJson);			
-		} catch (IOException e) {
-			throw new RuntimeException(e);
-		}
-	}
-	
-	private JMeterFunction buildTestFunction() {
-		File file = ResourceExtractor.extractResource(this.getClass().getClassLoader(), "scripts/Demo_JMeter.jmx");
-		JMeterFunction f = new JMeterFunction();
-		
-		f.setJmeterTestplan(new DynamicValue<String>(file.getAbsolutePath()));
 
-		//f.setLibrariesFile(new DynamicValue<>());
-		f.setId(new ObjectId());
-		Map<String, String> attributes = new HashMap<>();
-		attributes.put(AbstractOrganizableObject.NAME, "medor");
-		f.setAttributes(attributes);
+    private static final Logger logger = LoggerFactory.getLogger(JMeterHandlerTest.class);
+    private String jMeterHome;
+
+    @Before
+    public void before() {
+        jMeterHome = System.getenv().get(JMeterFunctionTypeLocalPlugin.JMETER_HOME_ENV_VAR);
+    }
+
+    @Category(step.junit.categories.LocalJMeter.class)

```

Comment:
@jeromecomte 
Yes, you are right. We wanted to ask @jeromebrongniart about proper configuration of maven profiles to enable/disable required categories in test. 
Now all builds on build server fail, because it is not configured yet.

Reply:
@jeromecomte @iegorov777 : sure, I will look at it

Reply:
I have it running on the build-test server but I got this error:
```
17:39:27,103 [ERROR] completeDescriptorReadTest(step.automation.packages.yaml.AutomationPackageDescriptorReaderTest)  Time elapsed: 5.717 s  <<< ERROR!
step.automation.packages.AutomationPackageReadingException: Unable to read the automation package yaml
	at step.automation.packages.yaml.AutomationPackageDescriptorReaderTest.completeDescriptorReadTest(AutomationPackageDescriptorReaderTest.java:74)
Caused by: step.plans.parser.yaml.schema.YamlPlanValidationException: #: extraneous key [scheduler] is not permitted
	at step.automation.packages.yaml.AutomationPackageDescriptorReaderTest.completeDescriptorReadTest(AutomationPackageDescriptorReaderTest.java:74)
Caused by: org.everit.json.schema.ValidationException: #: extraneous key [scheduler] is not permitted
	at step.automation.packages.yaml.AutomationPackageDescriptorReaderTest.completeDescriptorReadTest(AutomationPackageDescriptorReaderTest.java:74)
```

https://build.stepcloud-test.ch/#/root/executions/65412aefd06922734c7710d3/steps?tenant=build

@iegorov777  Any idea?

Reply:
@jeromebrongniart 
Yes, it is my fault. Fixed now.